### PR TITLE
opendevin/core/main.py: Graceful shutdown

### PR DESCRIPTION
**Pull request by li-boxuan**

### CODE REVIEW

File: opendevin/core/main.py [hunk 1]
```diff
@@ -26,7 +26,10 @@ def read_task_from_stdin() -> str:
 
 
 async def main(task_str: str = ''):
```
Comment (neubig) [inline opendevin/core/main.py:28]: This change modifies the return type of this function to `None`, so just to be safe could you add typing info? 

```suggestion
async def main(task_str: str = '') -> None:
```

File: opendevin/core/main.py [hunk 2]
```diff
@@ -94,7 +97,7 @@ async def on_event(event: Event):
     ]:
         await asyncio.sleep(1)  # Give back control for a tick, so the agent can run
 
-    return controller
+    await controller.close()
```
Comment (yufansong) [inline opendevin/core/main.py:100]: Why return `.close()`

Comment (neubig) [reply opendevin/core/main.py:100]: (it's not returning, just awaiting)

Comment (yufansong) [reply opendevin/core/main.py:100]: Oh, may bad. Then that is ok.

**Review by rbren — APPROVED** [review]
I bet this is where all my zombie sandboxes are coming from!

**Review by yufansong — APPROVED** [review]

**Pull request merged**